metagraph/src/graph/graph_extensions/node_rc.cpp

```diff
@@ -229,6 +230,60 @@ void NodeRC::call_incoming_from_rc(node_index node,
     }
 }
 
+void NodeRC::call_outgoing_from_rc(node_index node,
+                                   const std::function<void(node_index, char)> &callback) const {
+    assert(graph_);
+
+    if (const auto *dbg_succ_ = dynamic_cast<const DBGSuccinct*>(graph_)) {
+        const auto &boss = dbg_succ_->get_boss();
+        adjacent_outgoing_from_rc(node, [&](node_index next) {
+            char c = boss.decode(boss.get_W(dbg_succ_->kmer_to_boss_index(next)) % boss.alph_size);
+
+            if (c == boss::BOSS::kSentinel)
+                return;
+
+            callback(next, complement(c));
+        });
+    } else {
+        adjacent_outgoing_from_rc(node, [&](node_index next) {
+            char c = graph_->get_node_sequence(next).back();
+            if (c == boss::BOSS::kSentinel)
+                return;
+
+            callback(next, complement(c));
+        });
+    }
+}
+
+void NodeRC::call_incoming_from_rc(node_index node,
+                                   const std::function<void(node_index, char)> &callback) const {
+    assert(graph_);
+
+    if (const auto *dbg_succ_ = dynamic_cast<const DBGSuccinct*>(graph_)) {
+        const auto &boss = dbg_succ_->get_boss();
+        auto first_cache = dbg_succ_->get_extension<NodeFirstCache>();
+        adjacent_incoming_from_rc(node, [&](node_index prev) {
+            char c = first_cache
+                ? first_cache->get_first_char(prev)
+                : boss.decode(boss.get_minus_k_value(
+                    dbg_succ_->kmer_to_boss_index(prev), boss.get_k() - 1).first);
+
+            if (c == boss::BOSS::kSentinel)
+                return;
+
+            callback(prev, complement(c));
+        });
+    } else {
+        adjacent_incoming_from_rc(node, [&](node_index prev) {
+            char c = graph_->get_node_sequence(prev)[0];
+            if (c == boss::BOSS::kSentinel)
+                return;
+
+            callback(prev, complement(c));
+        });
+    }
+}
```

Comment:
I suggest reimplementing this. That is, remove the calls to `adjacent_incoming_from_rc`.
Also, use `NodeFirstCache::call_incoming_kmers`?

Reply:
done